lib/Parse/Lexer.cpp

```diff
@@ -613,7 +613,20 @@ void Lexer::lexIdentifier() {
   // Lex [a-zA-Z_$0-9[[:XID_Continue:]]]*
   while (advanceIfValidContinuationOfIdentifier(CurPtr, BufferEnd));
 
-  tok Kind = kindOfIdentifier(StringRef(TokStart, CurPtr-TokStart), InSILMode);
+  auto Str = StringRef(TokStart, CurPtr-TokStart);
+  tok Kind = kindOfIdentifier(Str, InSILMode);
+
+  // We don't want to support "isnt" without an apostrophe
+  if (Kind == tok::kw_isnt) {
+    Kind = tok::identifier;
+  }
+
+  if (Kind == tok::identifier && Str == "isn" &&
```

Comment:
This creates a lexing ambiguity with the planned `ain't` keyword, and the `whomst'd've'ly'yaint'nt'ed'ies's'y'es` feature.